### PR TITLE
Todo card creation ui

### PR DESCRIPTION
**Pull request by DDKatch**
screen shot 2018-11-27 at 2 35 29 pm

screen shot 2018-11-27 at 2 18 50 pm

UPD:
screen shot 2018-11-28 at 12 01 16 pm

screen shot 2018-11-28 at 12 01 42 pm

### CODE REVIEW

**Comment by DDKatch** [comment]
[I as user want to add new TODO card](https://trello.com/c/iilYjoHc/9-i-as-user-want-to-add-new-todo-card)

File: client/src/containers/CreateCardFormContainer/index.js
Comment (AleksSenkou) [inline client/src/containers/CreateCardFormContainer/index.js]: @DDKatch community naming for such events are with preposition `on...`

So:

* `onChange ` instead of `handleChange`
* `onSubmit` instead of `handleSubmit`

File: client/src/containers/CreateCardFormContainer/index.js
Comment (AleksSenkou) [inline client/src/containers/CreateCardFormContainer/index.js]: @DDKatch thoughts about giving handleChange and handleSubmit names related to business logic?

* `updateDescription` instead of `handleChange`
* `createCard` instead of `handleSubmit`

File: client/src/containers/CreateCardFormContainer/index.js
Comment (AleksSenkou) [inline client/src/containers/CreateCardFormContainer/index.js]: useless

File: client/.eslintrc
Comment (AleksSenkou) [inline client/.eslintrc]: 👍

File: client/src/components/CreateCardForm/index.js
Comment (AleksSenkou) [inline client/src/components/CreateCardForm/index.js]: @datarockets/cuties thoughts about using `'` or `"` everywhere?

File: client/src/components/CreateCardForm/index.js
Comment (AleksSenkou) [inline client/src/components/CreateCardForm/index.js]: image

Thoughts?

File: client/src/containers/CreateCardFormContainer/index.js
Comment (AleksSenkou) [inline client/src/containers/CreateCardFormContainer/index.js]: We can go with autobinding for that: https://medium.com/komenco/react-autobinding-2261a1092849

Can be used:
1) autobind decorator
```js
import autobind from 'autobind-decorator'

@autobind
handleChange(event) {
  this.setState({ description: event.target.value })
}
```
2) arrow function

```js
handleChange = (event) => {
  this.setState({ description: event.target.value })
}
```

instead of:
```js
this.handleChange = this.handleChange.bind(this)
```
in constructor

**Review by AleksSenkou — COMMENTED** [review]
Comments and suggestions over here
UI looks good 👍

**Comment by AleksSenkou** [comment]
@DDKatch I'm missing endpoints discussion for this client code. Where can I find it? We can go in trello card

**Comment by AleksSenkou** [comment]
After installing new modules locally I have changes in yarn.lock

image


Is it ok?

File: client/src/components/CreateCardForm/index.js
Comment (DDKatch) [reply client/src/components/CreateCardForm/index.js]: @AleksSenkou I don't have this error. What have you done?

**Comment by DDKatch** [comment]
> After installing new modules locally I have changes in yarn.lock
> 
> image
> 
> Is it ok?

@AleksSenkou there were creepy errors in yarn lock after rebase, so I remove it and regenerate.

**Comment by AleksSenkou** [comment]
@DDKatch do we need to regenerate it locally?

File: client/src/components/CreateCardForm/index.js
Comment (AleksSenkou) [reply client/src/components/CreateCardForm/index.js]: @DDKatch right after starting the dev server

Comment (DDKatch) [reply client/src/components/CreateCardForm/index.js]: @AleksSenkou let's do a PP for that, I can't figure out what's wrong.

Comment (AleksSenkou) [reply client/src/components/CreateCardForm/index.js]: @DDKatch firstly, let's try to figure out by chat. Do you have this error on server start?
@nikitasakov do you?

**Comment by DDKatch** [comment]
@AleksSenkou I don't know but it said that there was smth wrong in some string, but It looked familiar with others, so I've just regenerated

File: client/src/components/CreateCardForm/index.js
Comment (DDKatch) [reply client/src/components/CreateCardForm/index.js]: @AleksSenkou I prefer "" because lots of boilerplate code and tutoruals use doublequotes

Comment (AleksSenkou) [reply client/src/components/CreateCardForm/index.js]: @nikitasakov what about you?

Comment (AleksSenkou) [reply client/src/components/CreateCardForm/index.js]: from my side: I'm ok with both

**Comment by AleksSenkou** [comment]
@DDKatch let us know after merging, so we can regenerate locally 😉

File: client/src/components/CreateCardForm/index.js
Comment (DDKatch) [reply client/src/components/CreateCardForm/index.js]: @AleksSenkou No, I don't

File: client/src/components/CreateCardForm/index.js
Comment (AleksSenkou) [reply client/src/components/CreateCardForm/index.js]: We can discuss it in parallel and create card for updating linter in order not to block the PR

Comment (AleksSenkou) [reply client/src/components/CreateCardForm/index.js]: https://trello.com/c/uZc2dbSs/24-when-i-develop-the-app-id-like-strings-to-be-consistent-across-the-app-so-id-like-to-use-single-double-quarters-everywhere

Leave your thoughts there

**Comment by DDKatch** [comment]
> @DDKatch do we need to regenerate it locally?

Just run `yarn`

File: client/src/containers/CreateCardFormContainer/index.js
Comment (DDKatch) [reply client/src/containers/CreateCardFormContainer/index.js]: @AleksSenkou should we rename this props with onChange or setDescription | onSubmit or createCard ?

**Comment by DDKatch** [comment]
@AleksSenkou what about your error? Does it still exist?

**Comment by AleksSenkou** [comment]
@DDKatch Yeah, I had it.

We've used `this.description` instead of `this.state.description` 😉 

I've checked it in React dev tools

File: client/src/containers/CreateCardFormContainer/index.js
Comment (AleksSenkou) [inline client/src/containers/CreateCardFormContainer/index.js]: Simpler format can be used:
```js
state = {
  description: '',
}
```

instead of:

```js
constructor(props) {
  super(props)
  this.state = {
    description: '',
  }
}
```

File: client/src/containers/CreateCardFormContainer/index.js
Comment (AleksSenkou) [inline client/src/containers/CreateCardFormContainer/index.js]: `this.description` returns `undefined` all the time, `this.state.description` should be used instead 😉

**Review by AleksSenkou — APPROVED** [review]
LGTM with suggestions

File: client/src/containers/CreateCardFormContainer/index.js
Comment (DDKatch) [reply client/src/containers/CreateCardFormContainer/index.js]: 👌

File: client/src/containers/CreateCardFormContainer/index.js
Comment (DDKatch) [reply client/src/containers/CreateCardFormContainer/index.js]: 👌

**Comment by nikitasakov** [comment]
LGTM :)

**Pull request merged**
